docs/architecture/benchmark_comparison_csv_parquet_memmap.md

```diff
@@ -0,0 +1,78 @@
+# Benchmark Report: CSV vs Parquet vs Memmap for OHLC Data Reads
+
+## Objective
+
+The objective of this benchmark is to evaluate and compare the read performance of three different OHLC data storage approaches used in our system:
+
+- **Partitioned CSV Provider**
+- **Partitioned Parquet Provider**
+- **Partitioned Memmap Provider**
+
+These storage formats are used to cache and serve historical OHLC (Open, High, Low, Close) financial data. Performance is critical due to high-frequency read operations required during backtesting and trading simulations.
+
+---
+
+## Test Environment
+
+- **Machine Specs**: 24 GB RAM
```

Comment:
Let's run tests for different environments too.
- Different resampling frequencies (2min, 10min, 1h, 3d, etc...). 
- Different file sizes. Small files (only 10 rows for example), Medium files (already done), Large files (2 years of data).
- Let's increase the consecutive reads. Even if the test takes a bit longer to run. At least 30 repetitions to make the sample statistically significant.